### ISSUE TITLE
im2txt: Why is the initial learning rate so high (2.0)?

### ISSUE THREAD

**Issue by danieljl** [tensorflow/models#1037]
Is it a bug or is it expected that high? https://github.com/tensorflow/models/blob/master/im2txt/im2txt/configuration.py#L93

**Comment by aselle** [comment]
Do you have any reason to believe it is a bug. Did you try it and find that it produced Nans?

**Comment by danieljl** [comment]
Yes. The training has been running for a day and the total loss has been stuck in a plateau since the forth hour. I suspect it's a bug also because typical values of learning rate are below 1.0.

**Comment by aselle** [comment]
@cshallue, could you comment please?

**Comment by cshallue** [comment]
2.0 is the correct value.

The total loss reported by the training script is over a single batch and is therefore very noisy. This is why it appears not to be going down, but it really is on average. To see training progress you should look at perplexity on the validation set by running the evaluation script in a separate process.
https://github.com/tensorflow/models/tree/master/im2txt#training-a-model

**Comment by danieljl** [comment]
@cshallue Thanks for the clarification! However, TensorBoard shows that the losses are not going down. Is it normal to have like this until 400k steps? I executed the training script exactly as suggested in README and didn't change any parameters. I'm using Tesla K80 if that helps.

![image](https://cloud.githubusercontent.com/assets/1947204/23153678/637d4382-f83c-11e6-820d-6a82c162938c.png)

**Comment by cshallue** [comment]
@danieljl This is working as expected. If you were to zoom in on the total_loss graphs with sufficient smoothing it would be steadily decreasing on average. If you want a better plot to show that the loss is decreasing, you need to look at perplexity over the validation set. To do this, run the evaluation script in a separate process.
https://github.com/tensorflow/models/tree/master/im2txt#training-a-model

**Comment by aselle** [comment]
Thanks @cshallue for the detailed response.

**Comment by danieljl** [comment]
@cshallue OK I'll try it then. Many thanks!

**Comment by vanpersie32** [comment]
@danieljl @cshallue  Hey, sorry to bother you. I know the im2txt model can be trained well with such high learning rate = 2.0. But I still have question why do you choose such high learning rate and why it works. In my common practice, I usually set learning to below 0.1 and such high learning rate = 2.0 would cause the loss  to never drop.
